### PR TITLE
Added TTS Mod

### PR DESCRIPTION
**Pull request by TheRealPikachu**
**Please describe the changes this PR makes and why it should be merged:**
Pair this mod with Join Voice Channel and Play URL and you can finally have a talking Discord bot.
Made by TheRealPikachu#0001 and Shoot.Dot#0641.
**Status**

- [ ] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: text_to_speech_MOD.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+module.exports = {
+  name: 'Text To Speech',
+  section: 'Messaging',
+  
+  subtitle: function(data) {
+    const info = ['Item 1', 'Item 2', 'Item 3'];
```
Comment (noahalma) [inline text_to_speech_MOD.js:6]: What is this line for? You don’t use `info` anywhere

Comment (TheRealPikachu) [reply text_to_speech_MOD.js]: removed

File: actions/text_to_speech_MOD.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+module.exports = {
+  name: 'Text To Speech',
+  section: 'Messaging',
+
+  subtitle: function (data) {
```
Comment (almeidx) [inline actions/text_to_speech_MOD.js:5]:
```suggestion
  subtitle (data) {
```

File: actions/text_to_speech_MOD.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return 'Make your Discord bot talk.'
+  },
+
+  variableStorage: function (data, varType) {
```
Comment (almeidx) [inline actions/text_to_speech_MOD.js:9]:
```suggestion
  variableStorage (data, varType) {
```

File: actions/text_to_speech_MOD.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    const type = parseInt(data.storage)
+    if (type !== varType) return
+    const dataType = 'Audio URL'
+    return ([data.varName, dataType])
+  },
+
+  fields: ['text', 'storage', 'varName'],
+
+  html: function (isEvent, data) {
```
Comment (almeidx) [inline actions/text_to_speech_MOD.js:18]:
```suggestion
  html (isEvent, data) {
```

File: actions/text_to_speech_MOD.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    ${data.variables[1]}
+  </select>
+</div>
+<div id="varNameContainer" style="float: right; width: 60%;">
+  Variable Name:<br>
+  <input id="varName" class="round" type="text">
+</div>`
+  },
+
+  init: function () { },
```
Comment (almeidx) [inline actions/text_to_speech_MOD.js:39]:
```suggestion
  init () {},
```

File: actions/text_to_speech_MOD.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  action: async function (cache) {
```
Comment (almeidx) [inline actions/text_to_speech_MOD.js:41]:
```suggestion
  async action (cache) {
```

File: actions/text_to_speech_MOD.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const text = this.evalMessage(data.text, cache)
+    const Mods = this.getMods()
+    const tts = Mods.require('google-tts-api')
+    const play = await tts(text, 'en', 1)
+    this.storeValue(play, storage, varName, cache)
+    this.callNextAction(cache)
+  },
+  mod: function () { }
```
Comment (almeidx) [inline actions/text_to_speech_MOD.js:52]:
```suggestion

  mod () {}
```

File: actions/text_to_speech_MOD.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    const storage = parseInt(data.storage)
+    const varName = this.evalMessage(data.varName, cache)
+    const text = this.evalMessage(data.text, cache)
+    const Mods = this.getMods()
+    const tts = Mods.require('google-tts-api')
+    const play = await tts(text, 'en', 1)
+    this.storeValue(play, storage, varName, cache)
+    this.callNextAction(cache)
+  },
+  mod () {}
```
Comment (almeidx) [inline actions/text_to_speech_MOD.js:52]:
```suggestion

  mod () {}
```

**Comment by almeidx** [comment]
Add this new action to the docs file

File: docs/mods.json [hunk 9]
```diff
@@ -755,6 +755,10 @@
     "description": "",
     "name": "Sync Channel Permissions"
   },
+  "text_to_speech_MOD.js": {
+    "description": "",
```
Comment (almeidx) [inline docs/mods.json:759]: Add the description to the mod?

**Comment by almeidx** [comment]
Not a proper description of it

File: docs/mods.json [hunk 10]
```diff
@@ -755,6 +755,10 @@
     "description": "",
     "name": "Sync Channel Permissions"
   },
+  "text_to_speech_MOD.js": {
+    "description": "Convert some text into speech (using Discord tts)",
```
Comment (almeidx) [inline docs/mods.json:759]:
```suggestion
    "description": "Convert some text into TTS",
```

File: actions/text_to_speech_MOD.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    const dataType = 'Audio URL'
+    return ([data.varName, dataType])
+  },
+
+  fields: ['text', 'storage', 'varName'],
+
+  html (isEvent, data) {
+    return `
+<div>
+  <p>Please pair this with Join Voice Channel & Play URL. Store Audio URL In stores an Audio URL. Please paste this in Play URL for full effect.</p>
```
Comment (almeidx) [inline actions/text_to_speech_MOD.js:21]:
```suggestion
  <p>This action can not deal with texts which are over than 200 characters.</p>
```

**Pull request merged**